### PR TITLE
Added MatchingRules for Tweet

### PR DESCRIPTION
**Pull request by marcogruhl**
Added MatchingRules for Tweet
Extend Examples with Matching Rules
Added Author Every3Minutes to examples because it tweets every 3 minutes
Corrected comment

### CODE REVIEW

File: src/Client/TwitterClient.cs [hunk 1]
```diff
@@ -82,10 +81,11 @@ private static void IncludesParseMedias(IHaveMedia[] data, Includes includes)
             {
                 IncludesParseMedias(m, includes);
             }
-        }
+        }
```
Comment (Xwilarg) [inline src/Client/TwitterClient.cs:84]:
```suggestion
        }
```

File: README.md [hunk 2]
```diff
@@ -104,7 +105,7 @@ Task.Run(async () =>
     // Since we want to get the basic info of the tweet author, we add an empty array of UserOption
     await client.NextTweetStreamAsync((tweet) =>
     {
-        Console.WriteLine($"From {tweet.Author.Name}: {tweet.Text}");
+         Console.WriteLine($"From {tweet.Author.Name}: {tweet.Text} (Rules: {string.Join(',', tweet.MatchingRules.Select(x => x.Tag))})");
```
Comment (Xwilarg) [inline README.md:108]: Is the empty space added normal?

File: src/Response/Answer.cs [hunk 3]
```diff
@@ -29,4 +30,10 @@ internal class Summary
     {
         public int Deleted { init; get; }
     }
+
+    public class MatchingRule
```
Comment (Xwilarg) [inline src/Response/Answer.cs:34]: Shouldn't this be internal?

File: src/Client/TwitterClient.cs [hunk 4]
```diff
@@ -102,6 +102,10 @@ private static void InternalIncludesParse<T>(Answer<T> answer)
                     IncludesParseMedias((IHaveMedia)data, answer.Includes);
                     answer.Data = data;
                 }
+                if (answer.MatchingRules != null && answer.MatchingRules.Length > 0 && _matchingRulesInterface.IsAssignableFrom(typeof(T)))
```
Comment (Xwilarg) [inline src/Client/TwitterClient.cs:105]: .Length > 0 can be replaced by .Any()

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -88,9 +88,10 @@ var request = new TwitterSharp.Request.StreamRequest(
             Expression.Author("takanashikiara"),
             Expression.Author("ninomaeinanis"),
             Expression.Author("gawrgura"),
+            Expression.Author("Every3Minutes"),
             Expression.Author("watsonameliaEN")
```
Comment (Xwilarg) [inline README.md:92]: This is correct per see but I feel like the mix between anime vtubers and slavery is a bit strange, maybe make it a separate rule?

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (marcogruhl) [reply README.md:92]: good point. I thought you chose them for a high tweet rate 😅. I check if a separate rule makes sense in the sample context. Or i will completely remove it.

File: README.md (hunk 2 above)
Comment (marcogruhl) [reply README.md:108]: Overseen this. IDE works not that well with md files. Will be fixed.

File: src/Response/Answer.cs (hunk 3 above)
Comment (marcogruhl) [reply src/Response/Answer.cs:34]: The type is used at Tweet.cs L83 as normal property. I could separate an internal and public Type. Don't know if that makes sense. I am not 100% aware of that. Folowed mostly your patterns.

File: src/Client/TwitterClient.cs (hunk 4 above)
Comment (marcogruhl) [reply src/Client/TwitterClient.cs:105]: Changed that. Copy pasted it from your code 😉

File: src/Response/Answer.cs (hunk 3 above)
Comment (Xwilarg) [reply src/Response/Answer.cs:34]: Ah my bad, didn't read properly, mayble make it a record instead of a class since it's immutable then?

Comment (marcogruhl) [reply src/Response/Answer.cs:34]: Or I make it a "normal" separate file like Attachments.cs and remove it from the answer file? And also make it a record ...

Comment (Xwilarg) [reply src/Response/Answer.cs:34]: Yeah I think it would be more logic

File: src/Response/RTweet/MatchingRule.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿namespace TwitterSharp.Response.RTweet
+{
+    public record MatchingRule
+    {
+        public string Id { init; get; }
+        public string Tag { init; get; }
```
Comment (Xwilarg) [inline src/Response/RTweet/MatchingRule.cs:6]: init specifier can be internal I think

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: src/Response/RTweet/MatchingRule.cs (hunk 6 above)
Comment (marcogruhl) [reply src/Response/RTweet/MatchingRule.cs:6]: It does not work. Tested it, and the values were empty. Don't know why. Maybe the deserialization works some kind of outside the assembly?

Comment (Xwilarg) [reply src/Response/RTweet/MatchingRule.cs:6]: Ah yup, that's possible

**Review by Xwilarg — APPROVED** [review]

**Comment by Xwilarg** [comment]
https://www.nuget.org/packages/TwitterSharp/1.4.0

**Pull request merged**